src/dms/frontend/EdFi.DataManagementService.Frontend.AspNetCore/Program.cs

```
@@ -165,6 +202,48 @@ async Task RetrieveAndCacheClaimSets(WebApplication app)
     }
 }
 
+async Task InitializeDmsInstances(WebApplication app)
+{
+    app.Logger.LogInformation("Loading DMS instances from Configuration Service");
+    try
+    {
+        var dmsInstanceProvider = app.Services.GetRequiredService<IDmsInstanceProvider>();
+
+        IList<DmsInstance> instances = await dmsInstanceProvider.LoadDmsInstances();
+
+        if (instances.Count == 0)
+        {
+            app.Logger.LogCritical(
+                "No DMS instances were loaded from Configuration Service. DMS cannot start without instance configuration."
+            );
+            Environment.Exit(-1);
```

It is not strictly necessary for DmsInstances to exist on DMS Startup, but for now we fail hard if there are none.